packages/expo/src/Updates/Updates.ts

```diff
@@ -1,9 +1,23 @@
 import { UnavailabilityError } from '@unimodules/core';
 import { EventEmitter, EventSubscription } from 'fbemitter';
 import DeviceEventEmitter from 'react-native/Libraries/EventEmitter/RCTDeviceEventEmitter';
+import Constants from 'expo-constants'
 
 import ExponentUpdates from './ExponentUpdates';
 
+type Manifest = typeof Constants.manifest
+
+interface UpdateEvent {
+  /* Type of the event */
+  type: 'downloadStart' | 'downloadProgress' | 'downloadFinished' | 'noUpdateAvailable' | 'error',
+  /* If `type === Expo.Updates.EventType.DOWNLOAD_FINISHED`, the manifest of the newly downlaoded update. Undefined otherwise. */
+  manifest?: Manifest,
+  /* If `type === Expo.Updates.EventType.ERROR`, the error message. Undefined otherwise. */
```

Comment:
Could we express this in types instead of comments?

```ts
type UpdateEvent =
  | { type: 'downloadStart' | 'downloadProgress' |  'noUpdateAvailable', manifest: undefined, message: undefined }
  | { type: 'downloadFinished', manifest: Manifest, message: undefined }
  | { type: 'error', manifest: undefined, message: string }
```